pages/models.py

```diff
@@ -548,64 +548,197 @@ def get_context(self, *args, **kwargs):
         ]
 
 
-
-## CANDIDATES
-
+# CANDIDATES
 class CandidateEndorsementPage(Page):
-    body = RichTextField()
-    candidate = models.ForeignKey('endorsements.Candidate', null=True, blank=True, on_delete=models.SET_NULL)
+    result_choices = (
+        ('win', 'Win'),
+        ('loss', 'Loss'),
+    )
+    result_max_length = 16
+
+    body = RichTextField(verbose_name="Bio")
+    candidate = models.ForeignKey(
+        'endorsements.Candidate',
+        help_text='Ignore - legacy field for old pages.',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL
+    )
+    donate_url = models.URLField(null=True, blank=True)
     election = models.ForeignKey(
         'endorsements.Election',
         null=True,
         blank=True,
         on_delete=models.SET_NULL
     )
-    signup_tagline = models.CharField(max_length=128, blank=True, null=True)
-    social_image = models.ForeignKey('wagtailimages.Image', null=True, blank=True, on_delete=models.SET_NULL, related_name='+')
-    parent_page_types = ['pages.CandidateEndorsementIndexPage']
+    facebook_url = models.URLField(null=True, blank=True)
+    general_election_date = models.DateField(null=True, blank=True)
```

Comment:
Should this be required? Are there scenarios where someone wouldn't have a general election?

Reply:
Not required because:
- we are supporting legacy records with these fields empty (most important)
- user can save draft in progress with some fields empty
- if candidate loses in primary we don't really need the general election date